app/pb/threads.proto

```diff
@@ -0,0 +1,78 @@
+syntax = "proto3";
+
+package org.couchers.api.threads;
+
+import "google/protobuf/timestamp.proto";
+
+service Threads {
+  // This is a secure service: a user needs to be authenticated and not jailed to call functions here, refer to auth.proto and jail.proto
+
+  // Discussion threads are organized as a tree with maximum three levels depth:
+  // thread1
+  // +- comment1
+  // |  +- reply1
+  // |  +- reply2
+  // +- comment2
+  //    +- reply3
+  //    +- reply4
+  // thread2
+  // +- comment3
+  // |  +- reply5
+  // |  +- reply6
+  // +- comment4
+  //    +- reply7
+  //    +- reply8
+
+  // The same API is used regardless of level. All posts are using the same id namespace.
+
+  // Get non-deleted posts belonging to a thread, sorted in reverse time order.
+  // Pagination is supported by the posted_before_thread_id and number fields.
+  rpc GetThread(GetThreadReq) returns (GetThreadRes);
+
+  // Post a new comment on a thread. The post can not be changed or deleted after posting.
+  // Throws INVALID_ARGUMENT if thread_id corresponds to a nesting level 3 reply, since no futher nesting is allowed.
+  rpc PostReply(PostReplyReq) returns (PostReplyRes);
+}
+
+message GetThreadReq {
+  int64 thread_id = 1;
+
+  // Maximum number of replies to return. Defaults to 1000.
+  int32 page_size = 2;
```

Comment:
`uint32` as it's a count

Reply:
https://google.aip.dev/158 says `int32` but ok.